koku/masu/util/ibm/download.py

```diff
@@ -0,0 +1,94 @@
+#
+# Copyright 2021 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""IBM Download Utils."""
+import logging
+
+from django.conf import settings
+from ibm_cloud_sdk_core import BaseService
+from ibm_cloud_sdk_core.api_exception import ApiException
+from ibm_cloud_sdk_core.authenticators import IAMAuthenticator
+from ibm_platform_services.common import get_sdk_headers
+
+from api.provider.models import Provider
+from masu.config import Config
+from masu.util.aws.common import copy_local_report_file_to_s3_bucket
+from masu.util.common import get_path_prefix
+
+LOG = logging.getLogger(__name__)
+
+
+class NoMoreReportPages(Exception):
+    pass
+
+
+def page_downloader_factory(credentials, data_source, month):
+    service_version = "v1"
+    service_name = "resource-instance-usage-reports"
+    authenticator = IAMAuthenticator(credentials.get("iam_token", ""))
```

Comment:
We probably want to try to wrap some of this in a try except statement to catch potential errors.

Reply:
Yes, that's a good idea. At least log an error message so it will be easy to debug!